### DISCUSSION TITLE
本地包功能增强

### DISCUSSION THREAD

**Discussion by ddkwork** [xmake-io/xmake#2347] [Ideas]
1. 请为本地包增加版本控制

2. add headerfiles如果嵌套目录多于10个，写目录太麻烦，可否增加一个copy指定文件夹到include的接口，不必考虑污染，因为很多工程没有充足的使用经验，只是为了快速集成，遇到问题方便快速重构。

3. 如果可能，请为本地包的包集增加按序build接口

4. build的时候如何指定arch？clion和vscode切换arch经常出bug，全局清理有时候都不好使

**Comment by waruqi** [comment]
头一回见到这么长的 title，问个问题都不好好问，我实在是懒得看下去

**Comment by ddkwork** [comment]
> 头一回见到这么长的 title，问个问题都不好好问，我实在是懒得看下去

Sorry，修改了

**Comment by xq114** [comment]
> add headerfiles如果嵌套目录多于10个，写目录太麻烦，可否增加一个copy指定文件夹到include的接口

add_headerfiles("include/(**.h)") 不行吗

**Comment by ddkwork** [comment]


**Reply by waruqi** [reply]
跟 add_deps 有啥关系。。本地包也走  add_requires/add_packages
